README.md

```diff
@@ -1,2 +1,110 @@
-# terraform-google-log-export
-A Cloud Foundation Toolkit Module: Opinionated Google Cloud Platform project creation and configuration with Shared VPC, IAM, APIs, etc.
+# Terraform Logsink Module
+
+## Requirements
+### Terraform plugins
+- [Terraform](https://www.terraform.io/downloads.html) 0.10.x
+- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) plugin v1.8.0
+
+### Configure a Service Account
+In order to execute this module you must have a Service Account with the following:
+
+#### Roles
+The service account with the following roles:
+- roles/logging.configWriter on the logsink project (to create logsinks)
+- roles/serviceAccount.admin on the logsink project (to create a service account for the log sinks)
+- roles/iam.admin on the logsink project (to grant permissions to the service account)
+- roles/serviceusage.admin on both the logsink and the destination project (to enable API for destinations)
+- roles/storage.editor on the destination project (if logsink destination is a Cloud Storage bucket)
+- roles/pubsub.editor on the destination project (if logsink destination is a Cloud Pub/Sub topic)
+- roles/bigquery.dataEditor on the destination project (if logsink destination is a BigQuery table)
+
+
+### Enable API's
+In order to operate with the Service Account you must activate the following API's on the base project where the Service Account was created:
+
+- Cloud Resource Manager API - cloudresourcemanager.googleapis.com
+- Cloud Billing API - cloudbilling.googleapis.com
+- Identity and Access Management API - iam.googleapis.com
+- Service Usage API - serviceusage.googleapis.com
+
+## Install
+
+### Terraform
+Be sure you have the correct Terraform version (0.10.x), you can choose the binary here:
+- https://releases.hashicorp.com/terraform/
+
+## Usage
+You can go to the `examples` folder to see all the use cases, however the usage of the module could be like this in your own `main.tf` file:
+
+```
+    locals {
+      credentials_file_path = "<PATH TO THE SERVICE ACCOUNT JSON FILE>"
+    }
+
+    provider "google" {
+      credentials = "${file(local.credentials_file_path)}"
+    }
+
+    module "logsink" {
+	source = "./"
+	name   = "my-logsink"
+        folder = "2165468435"
+        filter = "severity >= ERROR"
+        include_children = true
+        pubsub = {
+          name    = "my-logsink-pubsub"
+          project = "my-pubsub-project"

```

Comment:
Could we add a "create" flag within the destination blocks which defaults to true? If it is false, it would assume the destination has already been created elsewhere and simply create the sink.

Reply:
Good idea ! Will do.